### PR TITLE
chore: minor nitpicking

### PR DESCRIPTION
**Pull request by michaelfig**
_Incidental_

## Description

Independent changes that are best reviewed commit-by-commit:
-  chore(fast-usdc): publish for `@agoric/builders` to use, since it is published to NPM and has a hard dependency on `@agoric/fast-usdc`
- test(e2e-tools): provide `proposal_id` and fix the "propsal [sic]" typo
- ci(lint-rest): move `lint-primary` doc generation to rebalance (since it was the longest-running non-integration ci job)
- ci(dump-ci-stats): bump action versions to use the same ones as other workflows, avoiding deprecation warnings

### Security Considerations

n/a

### Scaling Considerations

Slightly faster CI times.

### Documentation Considerations

n/a

### Testing Considerations

n/a

### Upgrade Considerations

n/a

### CODE REVIEW

File: .github/workflows/test-all-packages.yml [hunk 1]
```diff
@@ -121,6 +115,24 @@ jobs:
         run: yarn run build:submissions
         working-directory: a3p-integration
 
+  # Verify that building the documentation works correctly.
+  lint-docs:
```
Comment (mhofman) [inline .github/workflows/test-all-packages.yml]: We need to add a new `lint-docs` required check on all protected branches, right?

**Review by mhofman — APPROVED** [review]
LGTM

File: .github/workflows/test-all-packages.yml (hunk 1 above)
Comment (michaelfig) [reply .github/workflows/test-all-packages.yml]: > on all protected branches

More accurately: on all protected branches that have merged the current PR.

Comment (turadg) [reply .github/workflows/test-all-packages.yml]: Would it work as part of lint-rest? Just to save the repo admin activity. There may be more lint refactoring and naming each job creates friction.

**Review by turadg — APPROVED** [review]
Non blocking but consider doing this in a way that doesn't require changing required checks config or making PRs have to merge this and rerun to get to green.

**Comment by michaelfig** [comment]
> consider doing this in a way that doesn't require changing required checks config [and] making PRs have to merge this and rerun to get to green.

I was considering not marking this as a required check for `master` until it's been on `master` long enough that the only PRs needing to merge it are those needing to update anyway.

File: .github/workflows/test-all-packages.yml (hunk 1 above)
Comment (michaelfig) [reply .github/workflows/test-all-packages.yml]: > Would it work as part of lint-rest?

OIC, yes I'll do that.

**Comment by turadg** [comment]
> I was considering not marking this as a required check for `master` until it's been on `master` long enough that the only PRs needing to merge it are those needing to update anyway.

IME some PRs get revived after a very long time. This repo has [252 open PRs](https://github.com/Agoric/agoric-sdk/pulls?q=is%3Apr+is%3Aopen+sort%3Acreated-asc) dating back 3 years. No matter what time window you adopt there's a risk of merges to master in which the tests are failing. 

lint-rest and lint-docs both take 4m. 0.5min of that is warmup so they may take less than 8 min combined. That's less than test-inter-protocol.

**Review by turadg — APPROVED** [review]

**Pull request merged**